src/Components/Bodies/CodeBodyUnversionized.tsx

```diff
@@ -53,44 +52,29 @@ class CodeBodyUnversionized extends Component<Props, ICode> {
     /**
      * Does a case distinction for all the catalogs and set the string ready for fetching
      * @param language
-     * @param resource_type
-     * @param version
-     * @param code
+     * @param resourceType ('migels', 'als', 'drugs')
+     * @param code ('all' if base code, (non-)terminal code else)
+     * @param catalog ('MIGEL', 'AL', 'DRUG')
      * @returns {Promise<null|any>}
      */
-    async fetchHelper(language, resource_type, version, code) {
-        resource_type = resource_type.toUpperCase();
-        if(code === "all") {
-            code = resource_type
-        }
-        if (code === "all" && code !== 'AL') {
-            return null
-        } else {
-            if (version === 'AL'){
-                resource_type = resource_type + "/" + resource_type;
-                code = '?show_detail=1'
-            }
-            return await fetch('https://search.eonum.ch/' + language + "/" + version + "/" + resource_type + "/" + code + "?show_detail=1")
-                .then((res) => {
-                    return res.json()
-                })
-        }
+    async fetchHelper(language, resourceType, code, catalog) {
+        let fetchString = ['https://search.eonum.ch', language, resourceType, catalog, code, "?show_detail=1"].join("/")
+        return await fetch(fetchString)
+            .then((res) => {
+                return res.json()
+            })
     }
 
     /**
      * Fetch the information from the backend and does a case distinction for all the catalogs
      * @returns {Promise<void>}
      */
     async fetchInformations() {
-        let newAttributes, versions;
-        if (this.props.params.catalog === "MIGEL") {
-            versions = 'migels'
-        }else if (this.props.params.catalog === "AL") {
-            versions = 'laboratory_analyses';
-        }else if (this.props.params.catalog === "DRUG") {
-            versions = 'drugs'
-        }
-        newAttributes = await this.fetchHelper(this.props.params.language, this.props.params.catalog, versions, this.props.params.code)
+        let newAttributes;
+        const {language, code, resource_type, catalog} = this.props.params;
+        let codeForFetch = code === 'all' ? catalog : code;
+        let resourceType = catalog === 'AL' ? 'laboratory_analyses' : resource_type;
```

Comment:
Or just use this expression directly as function argument. This will reduce the possible error of confounding the two variables that are very similar.